src/query/interpret/eval.hpp

```diff
@@ -903,7 +903,14 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
     return TypedValue(std::move(result), ctx_->memory);
   }
 
-  TypedValue Visit(Exists &exists) override { return TypedValue{frame_->at(symbol_table_->at(exists)), ctx_->memory}; }
+  TypedValue Visit(Exists &exists) override {
+    if (frame_->at(symbol_table_->at(exists)).IsFunction()) [[likely]] {
```

Comment:
I would cache accesses to symbol_table and frame_

Reply:
Okay, will do